test/modules/http/intercept/http.request.test.ts

```diff
@@ -230,6 +230,30 @@ it('intercepts an http.request given RequestOptions without a protocol', async (
   expect(requestId).toMatch(UUID_REGEXP)
 })
 
+it('intercepts an http.request path in url and options', async () => {
```

Comment:
I've left only one of these actual request tests since they are quite verbose and assert many things apart from the correct request options. I've moved the majority of tests to the unit test for `normalizeClientRequestArgs`, which also allowed us to cover more use cases.